packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -523,15 +524,15 @@ function validateClassAttr(vnode: VBaseElement, elm: Element, renderer: Renderer
             }
         }
 
-        readableVnodeClassname = computedClassName.trim();
+        readableVnodeClassname = JSON.stringify(computedClassName.trim());
 
         if (classList.length > keys(classMap).length) {
             nodesAreCompatible = false;
         }
-    } else if (isUndefined(className) && elmClassName !== '') {
+    } else if (isUndefined(className) && !isNull(elmClassName)) {
         // SSR contains a className but client-side VDOM does not
         nodesAreCompatible = false;
-        readableVnodeClassname = '';
+        readableVnodeClassname = '""';
```

Comment:
Existing error message:
`attribute "class" has different values, expected "" but found`
New error message:
`attribute "class" has different values, expected """" but found`

Reply:
I don't think that is the case. `readableVnodeClassname` is passed directly into the logged string:

https://github.com/salesforce/lwc/blob/ff8e24ca765c3d5de6812d4039b7eb75785f3adc/packages/%40lwc/engine-core/src/framework/hydration.ts#L540-L548

In fact, there is a test that covers this case. Note it says `""`, not `""""`:

https://github.com/salesforce/lwc/blob/ff8e24ca765c3d5de6812d4039b7eb75785f3adc/packages/%40lwc/integration-karma/test-hydration/mismatches/class-attr/only-present-in-ssr/index.spec.js#L14-L16

Reply:
Ah you are right. I was testing with the template literal in master branch and not your feature branch.